artiq/coredevice/comm_analyzer.py

```diff
@@ -180,9 +192,13 @@ async def _receive_cr(self):
                 # The remaining header length is 11 bytes.
                 remaining_data = await self.reader.readexactly(payload_length + 11)
                 data = endian_byte + payload_length_word + remaining_data
+                logger.info("received {} bytes from analyzer proxy".format(payload_length))
                 self.receive_cb(data)
-        finally:
+        except asyncio.IncompleteReadError:
```

Comment:
Receiver waits for a new dump on line 179
```
endian_byte = await self.reader.readexactly(1)
```
When closing the receiver, since the reader expects 1 byte it throws a `IncompleteReadError`. Since it is on close and any data received by the dashboard can be disregarded, the exception is ignored.

Reply:
> What do you mean by "closing the receiver"?

Exiting the dashboard.

Reply:
> Can you just use asyncio task cancellation like everywhere else?

I'll tinker with it a bit more and see if I can do that.